code/modules/mob/living/carbon/xenomorph/Xenomorph.dm

```diff
@@ -347,157 +347,159 @@
 	var/atom/movable/vis_obj/xeno_wounds/wound_icon_holder
 	var/atom/movable/vis_obj/xeno_pack/backpack_icon_holder
 
-/mob/living/carbon/xenomorph/Initialize(mapload, mob/living/carbon/xenomorph/oldXeno, h_number)
-	var/area/A = get_area(src)
-	if(A && A.statistic_exempt)
-		statistic_exempt = TRUE
+/mob/living/carbon/xenomorph/Initialize(mapload, mob/living/carbon/xenomorph/old_xeno, hivenumber)
+
+	if(old_xeno && old_xeno.hivenumber)
+		src.hivenumber = old_xeno.hivenumber
+	else if(hivenumber)
+		src.hivenumber = hivenumber
+
+	var/datum/hive_status/hive = GLOB.hive_datum[src.hivenumber]
+
+	if(hive)
+		hive.add_xeno(src)
 
 	wound_icon_holder = new(null, src)
 	vis_contents += wound_icon_holder
 
-	if(oldXeno)
-		set_movement_intent(oldXeno.m_intent)
-		hivenumber = oldXeno.hivenumber
-		nicknumber = oldXeno.nicknumber
-		life_kills_total = oldXeno.life_kills_total
-		life_damage_taken_total = oldXeno.life_damage_taken_total
-		evolution_stored = oldXeno.evolution_stored
-		if(oldXeno.iff_tag)
-			iff_tag = oldXeno.iff_tag
-			iff_tag.forceMove(src)
-			oldXeno.iff_tag = null
-	else if (h_number)
-		hivenumber = h_number
-
 	set_languages(list(LANGUAGE_XENOMORPH, LANGUAGE_HIVEMIND))
-	if(oldXeno)
-		for(var/datum/language/L in oldXeno.languages)
-			add_language(L.name)//Make sure to keep languages (mostly for event Queens that know English)
 
-	// Well, not yet, technically
-	var/datum/hive_status/in_hive = GLOB.hive_datum[hivenumber]
-	if(in_hive)
-		in_hive.add_xeno(src)
-		// But now we are!
+	///Handle transferring things from the old Xeno if we have one in the case of evolve, devolve etc.
+	if(old_xeno)
+		src.nicknumber = old_xeno.nicknumber
+		src.life_kills_total = old_xeno.life_kills_total
+		src.life_damage_taken_total = old_xeno.life_damage_taken_total
+		src.evolution_stored = old_xeno.evolution_stored
 
-	for(var/T in in_hive.hive_inherant_traits)
-		ADD_TRAIT(src, T, TRAIT_SOURCE_HIVE)
+		for(var/datum/language/language in old_xeno.languages)
+			add_language(language.name)//Make sure to keep languages (mostly for event Queens that know English)
+
+		//Carry over intents & targeted limb to the new Xeno
+		set_movement_intent(old_xeno.m_intent)
+		a_intent_change(old_xeno.a_intent)
+
+		//We are hiding, let's keep hiding if we can!
+		if(old_xeno.layer == XENO_HIDING_LAYER)
+			for(var/datum/action/xeno_action/onclick/xenohide/hide in actions)
+				if(istype(hide))
```

Comment:
the loop above automatically performs this typecheck, so this is redundant

Reply:
oh yea missed that one